src/codeu/chat/codeU_db/DataBaseConnection.java

```diff
@@ -12,7 +12,7 @@
 public final class DataBaseConnection {
 
   private Connection c = null;
-  private Statement stmt = null;
+  private PreparedStatement stmt = null;
```

Comment:
Do you still need this member variable? It looks like most of the functions below overwrite this and then close the statement. I think it would be better to just update your "stmt = c.prepareStatement()" code everywhere below to "PreparedStatement stmt = c.prepareStatement()" so people don't think the statements really need to live beyond the scope of the function.